Docs for synthetic source (#87416)
elastic/elasticsearch: docs/reference/mapping/types/text.asciidoc

    Metadata about the field.

ifeval::["{release-state}"=="unreleased"]
[[text-synthetic-source]]
==== Synthetic source

> `text` fields support <<synthetic-source,synthetic `_source`>> if they have
> a `keyword` sub-field that supports synthetic `_source` and *doesn't* have
> <<copy-to,`copy_to`>>.
> 
> Synthetic source will always sort `keyword` fields and remove duplicates so
> it will also do that for `text` fields using their sub-`keyword` field. So:

Reviewer: ```suggestion
`text` fields support <<synthetic-source,synthetic `_source`>> if they have
a `keyword` sub-field that supports synthetic `_source` and *do not* have
<<copy-to,`copy_to`>>.

Synthetic source always sorts `keyword` fields and removes duplicates, so
for `text` fields it sorts based on the sub-`keyword` field. For example:
```

    Metadata about the field.

ifeval::["{release-state}"=="unreleased"]
[[text-synthetic-source]]
==== Synthetic source
`text` fields support <<synthetic-source,synthetic `_source`>> if they have
a `keyword` sub-field that supports synthetic `_source` and *do not* have
<<copy-to,`copy_to`>>.

Synthetic source always sorts `keyword` fields and removes duplicates, so
`text` fields are sorted based on the sub-`keyword` field. For example:
[source,console,id=synthetic-source-text-example]
----
PUT idx
{
  "mappings": {
    "_source": { "synthetic": true },
    "properties": {
      "text": {
        "type": "text",
        "fields": {
          "raw": {
            "type": "keyword"
          }
        }
      }
    }
  }
}
PUT idx/_doc/1
{
  "text": [
    "the quick brown fox",
    "the quick brown fox",
    "jumped over the lazy dog"
  ]
}
----
// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]

Will become:
[source,console-result]
----
{
  "text": [
    "jumped over the lazy dog",
    "the quick brown fox"
  ]
}
----
// TEST[s/^/{"_source":/ s/\n$/}/]

> NOTE: Reordering text fields can have an effect on <<query-dsl-match-query-phrase,phrase>>
>       and <<span-queries,span>> queries. See the discussion about
>       <<position-increment-gap, here>> for more. You can avoid this by making sure the
>       `slop` parameter on the phrase queries is lower than the `position_increment_gap`.
>       This is the default.

Reviewer: ```suggestion
NOTE: Reordering text fields can have an effect on <<query-dsl-match-query-phrase,phrase>>
      and <<span-queries,span>> queries. See the discussion about
      <<position-increment-gap,`position_increment_gap`>> for more detail. You
      can avoid this by making sure the `slop` parameter on the phrase queries
      is lower than the `position_increment_gap`. This is the default.
```
Just a small fix for that link.
